### ISSUE TITLE
modulo() function doesn't work for negative numbers

### ISSUE THREAD

**Issue by saig0** [camunda/feel-scala#152]
**Describe the bug**
The `modulo()` function produces a wrong result for negative numbers.

**To Reproduce**
Expression: `modulo(10, -4)` returns `2` but should be `-2`.

**Expected behavior**
```
modulo(10, -4) -> -2
modulo(-10, 4) -> 2
modulo(-12, 5) -> -2
modulo(12, -5) -> 2
modulo(-10.1, 4.5) -> 3.4
modulo(10.1, -4.5) -> -3.4
```

**Environment**
* FEEL engine version: 1.11.2

**Comment by kaaquist** [comment]
I also have a fix for this here. 
Hope that is ok too?

**Comment by saig0** [comment]
From the DMN 1.3 spec:

![image](https://user-images.githubusercontent.com/4305769/94768974-364c5900-03b1-11eb-9da5-6242938a0dac.png)

**Comment by kaaquist** [comment]
Thanks. I overlooked that part.. So I need to use the DMN specs, just trying to clarify because the above state that `Expected behavior`:  
```
modulo(10, -4) -> -2
modulo(-10, 4) -> 2
modulo(-12, 5) -> -2
modulo(12, -5) -> 2
modulo(-10.1, 4.5) -> 3.4
modulo(10.1, -4.5) -> -3.4
```
That does not seem to be DMN ? 
I will change the code. 
:+1:

**Comment by saig0** [comment]
Yes. I just wrote the sample FEEL expressions. I took the expressions from the DMN TCK:

Expressions: https://github.com/dmn-tck/tck/blob/master/TestCases/compliance-level-3/0056-feel-modulo-function/0056-feel-modulo-function.dmn
Expected Results: https://github.com/dmn-tck/tck/blob/master/TestCases/compliance-level-3/0056-feel-modulo-function/0056-feel-modulo-function-test-01.xml

We should cover the test cases from the spec and TCK :)

**Comment by kaaquist** [comment]
I have created a fix. :smile:

**Comment by saig0** [comment]
It is merged. Nice job :cake: 

I'm interested. This is your second PR that is merged here. What is your background (e.g. using FEEL expressions in Camunda's DMN engine, using FEEL expressions in Zeebe, etc.) and your motivation for contributing?

**Comment by kaaquist** [comment]
I really like `scala` as a programming language, and I don't want to lie but I did find your project because I was looking for Issues I could fix as part of the `hacktoberfest`. Open source has always been something I really would like to be part of, but never go around too. So I did not really know anything about Camunda before hand. But I have looked through the website and saw the video etc. about the product. In my daily work I work with automation as a data engineer/cloud engineer using terraform, helm, helmfile etc. to setup data pipeline architecture around tracking data form websites/emails/sms/etc. 
So after looking trough Camunda site, I did actually think that could be something to look into, to be able to show our customers in an a better way the BPM of what take place. :smile:
